### PR TITLE
Start of notebook tutorials

### PR DESCRIPTION
**Pull request by mvkovatch3**
[ctdcal's ReadTheDocs](https://ctdcal.readthedocs.io/en/latest/) has had a lone Jupyter notebook converted to HTML for a while now. This PR adds a second notebook, both of which now actually use some `ctdcal` functionality.

These are intended as starting points for the tutorials to expand significantly as the non-ODF `ctdcal` user experience continues to improve.

### CODE REVIEW

File: ctdcal/io.py [hunk 1]
```diff
@@ -34,6 +56,99 @@ def load_exchange_btl(btl_file: Union[str, Path]) -> pd.DataFrame:
...
+        file = data_raw.read().decode("utf8").splitlines(keepends=True)
+
+    # process metadata
+    for idx, line in enumerate(file):
+        # find header info
+        if line.startswith("NUMBER_HEADERS"):
+            header_ind = idx
+
+        # find index of units row
+        if line.startswith("CTDPRS"):
```
Comment (DocOtak) [inline ctdcal/io.py:131]: Careful of two things here:
1. ordering of columns is not guaranteed, just "preferred"
2. There can be spaces in front of the param name

Comment (mvkovatch3) [reply ctdcal/io.py:131]: Would replacing this with 

```
if "CTDPRS" in line:  # for CTD

if "EXPOCODE" in line:  # for bottle
``` 

be sufficient? I'm pretty sure those are both required columns, right?
<hr>
edit:

If I also add this to the metadata loop
```
# skip comment lines
if line.strip().startswith("#"):
    continue
```

then it passes my tests.

Comment (mvkovatch3) [reply ctdcal/io.py:131]: Hm, this breaks if any comment line has either of those words in it... back to the drawing board.

Comment (mvkovatch3) [reply ctdcal/io.py:131]: Resolved with e93c72d0cae57c70258cebe52f5e40b2d7426575

Comment (DocOtak) [reply ctdcal/io.py:131]: Yeah I think that will do it.

The CCHDO code tends to actually use that NUMBER_HEADERS thing for alignment, we want our code to throw when it isn't correct (so we know to fix it).

**Pull request merged**